components/builder-events/src/event.rs

```diff
@@ -53,3 +53,22 @@ impl BuilderEvent {
 
     pub fn routing_key(self) -> Option<String> { self.routing_key }
 }
+
+#[macro_export]
+macro_rules! publish_event {
+    ($state:expr, $type:expr, $json:expr) => {
+        publish_event!($state, $type, $json, None)
+    };
+    ($state:expr, $type:expr, $json:expr, $routing_key:expr) => {
+        if let Some(bus) = $state.eventbus.as_ref() {
+            let event_type = $type;
```

Comment:
I'd just rename `$type` to `$event_type` in the parameter list and remove this line altogether.

Reply:
Modified in https://github.com/habitat-sh/builder/pull/1540/commits/dab949e9cc2965ca859a5ba57eb9c39bb158a1cb